File: python/head.py
import os

home_dir = os.path.expanduser("~")

desktop_dir = os.path.join(home_dir, "Desktop")

file_path = os.path.join(desktop_dir, 'Captura.png')


def file_head(file_name, size):
    with open(file_name, 'rb') as file:
        data = file.read(size)

    if len(data) < size:
        raise ValueError(f'{file_name!r} too small')

    return data


if __name__ == '__main__':
    breakpoint()
    data = file_head(file_path, 8)
    print(data)

File: go/head.go
package main

import (
	"fmt"
	"os"
)

func fileHead(fileName string, size int) ([]byte, error) {
	file, err := os.Open(fileName)

	if err != nil {
		return nil, err
	}

	defer file.Close()

	buffeer := make([]byte, size)
	n, err := file.Read(buffeer)

	if err != nil {
		return nil, err
	}

	if n != size {
		return nil, fmt.Errorf("%q too small", fileName)
	}

	return buffeer, nil
}
